test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1

```diff
@@ -5,219 +5,460 @@
+            }
+        )
+
+    foreach ($testCase in $testCases)
+    {
+        RunEncodingTestCase $testCase
+    }
+}
 
-        $result = 65..68 | Format-Hex
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        $actualResult = $result.ToString()
-        # whitespace sensitive
-        $actualResult | should be "00000000   41 42 43 44                                      ABCD            "
+    Context "Correct Expected Errors" {
+
+        $otherProvider = Get-ChildItem Cert:\CurrentUser\My\
+        $thumbprint = $otherProvider[0].Thumbprint
+
+        $testDirectory = $inputFile1.DirectoryName
+
+        $testCases = @(
+            @{
+                Name = "Does not support non-FileSystem Provider paths 'fhx -Path 'Cert:\CurrentUser\My\`$thumbprint' -ErrorAction Stop'"
+                PathParameterErrorCase = $true
+                Path = "Cert:\CurrentUser\My\$thumbprint"
+                expectedError = "FormatHexOnlySupportsFileSystemPaths,Microsoft.PowerShell.Commands.FormatHex"
+            }
+            @{
+                Name = "If path resolves to multiple files 'fhx -Path '`$testDirectory\*' -ErrorAction Stop'"
+                PathParameterErrorCase = $true
+                Path = "$testDirectory\*"
+                expectedError = "FormatHexResolvePathError,Microsoft.PowerShell.Commands.FormatHex"
+            }
+            @{
+                Name = "Type Not Supported 'fhx -InputObject @{'hash' = 'table'} -ErrorAction Stop'"
+                InputObjectErrorCase = $true
+                Path = $inputFile1
+                InputObject = @{ "hash" = "table" }
+                expectedError = "FormatHexTypeNotSupported,Microsoft.PowerShell.Commands.FormatHex"
+            }
+        )
+
+         foreach ($testCase in $testCases)
+        {
+            RunExpectedErrorTestCase $testCase
+        }
     }
 
-    # This test is to validate that integers can be piped to the format-hex
-    It "ValidateIntegerRawInput" {
+    Context "Continues to Process Valid Paths" {
 
-        $result = 1,2,3,4 | Format-Hex -Raw
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        $actualResult = $result.ToString()
-        # whitespace sensitive
-        $actualResult | should be "00000000   01 00 00 00 02 00 00 00 03 00 00 00 04 00 00 00  ................"
-    }
+        It "If given invalid path in array, continues to process valid paths 'fhx -Path `$invalidPath, `$inputFile1  -ev e -ErrorAction SilentlyContinue'" {
 
-    # handle int64
-    It "ValidateInteger64" {
+            $invalidPath = "$($inputFile1.DirectoryName)\fakefile8888845345345348709.txt"
+            $invalid2 = "C:\Windows\fakefile99342039847.txt"
+            $a = $null
```

Comment:
Could you please use a more meaningful variable name? E.g., $output, and $errorThrown?